service/history/workflow/update/registry.go

```diff
@@ -78,6 +78,7 @@ type (
 		GetAcceptedWorkflowExecutionUpdateIDs(context.Context) []string
 		GetUpdateInfo(ctx context.Context, updateID string) (*persistencespb.UpdateInfo, bool)
 		GetUpdateOutcome(ctx context.Context, updateID string) (*updatepb.Outcome, error)
+		GetUpdatesCount(ctx context.Context) int
```

Comment:
These update-specific calls are multiplying. Let's talk separately about if we can keep this interface tighter. Not for this PR though.